src/types/interfaces/searchfilter.interface.ts

```diff
@@ -0,0 +1,3 @@
+export interface SearchFilter {
+  [key: string]: string | null;
+}
```

Comment:
Can be replaced by Record<string, string | null>.
(https://www.typescriptlang.org/docs/handbook/advanced-types.html#mapped-types - described here)

Reply:
doesn't it make code more cumbersome?
also, that site says that this method has been deprecated, should I use it?

Reply:
Here is more direct link
https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype